File: app/src/main/java/com/roix/mvvm_archtecture_sample/utils/binding/DataBinder.java
package com.roix.mvvm_archtecture_sample.utils.binding;

import android.databinding.BindingAdapter;
import android.databinding.BindingConversion;
import android.support.annotation.ColorRes;
import android.support.annotation.DrawableRes;
import android.support.annotation.NonNull;
import android.support.v4.content.ContextCompat;
import android.view.View;
import android.widget.ImageView;

/**
 * Created by roix on 05.12.2017.
 */

public class DataBinder {
    @BindingConversion
    public static int convertBooleanToVisibility(final boolean visible) {
        return visible ? View.VISIBLE : View.GONE;
    }
    @BindingAdapter("bind:srcVector")
    public static void setSrcVector(@NonNull final ImageView imageView, @DrawableRes final int res) {
        imageView.setImageResource(res);
    }

    @BindingAdapter("bind:srcCompat")
    public static void setSrcCompat(@NonNull final ImageView imageView, @DrawableRes final int res) {
        imageView.setImageResource(res);
    }
    @BindingAdapter("bind:tint")
    public static void setTint(final ImageView view, @ColorRes final int colorRes) {
        view.setColorFilter(ContextCompat.getColor(view.getContext(), colorRes));
    }

    @BindingAdapter("bind:tintColor")
    public static void setTintColor(final ImageView view, final int color) {
        view.setColorFilter(color);
    }

    @BindingAdapter("bind:colorInt")
    public static void setColorInt(final ImageView view, final int color) {
        view.setColorFilter(color);
    }

}

File: app/src/main/java/com/roix/mvvm_archtecture_sample/utils/binding/DataBinder.kt
package com.roix.mvvm_archtecture_sample.utils.binding

import android.databinding.BindingAdapter
import android.databinding.BindingConversion
import android.support.annotation.ColorRes
import android.support.annotation.DrawableRes
import android.support.v4.content.ContextCompat
import android.view.View
import android.widget.ImageView

/**
 * Created by roix on 05.12.2017.
 */
@BindingConversion
fun convertBooleanToVisibility(visible: Boolean): Int {
    return if (visible) View.VISIBLE else View.GONE
}

@BindingAdapter("bind:srcVector")
fun setSrcVector(imageView: ImageView, @DrawableRes res: Int) {
    imageView.setImageResource(res)
}

@BindingAdapter("bind:srcCompat")
fun setSrcCompat(imageView: ImageView, @DrawableRes res: Int) {
    imageView.setImageResource(res)
}

@BindingAdapter("bind:tint")
fun setTint(view: ImageView, @ColorRes colorRes: Int) {
    view.setColorFilter(ContextCompat.getColor(view.context, colorRes))
}

@BindingAdapter("bind:tintColor")
fun setTintColor(view: ImageView, color: Int) {
    view.setColorFilter(color)
}

@BindingAdapter("bind:colorInt")
fun setColorInt(view: ImageView, color: Int) {
    view.setColorFilter(color)
}
